### DISCUSSION TITLE
"use-long-press" successfully triggers even with movement, but then does not finish when released

### DISCUSSION THREAD

**Discussion by derekantrican** [minwork/react#41] [Q&A]
I'm trying to build a small app that will change content as long as the finger is held down but will immediately change back when released. Here's the options I'm using:
```js
const bind = useLongPress(() => {
  console.log('long pressed');
  setTemporarySubs();
}, {
  threshold: 3000,
  cancelOutsideElement: false,
  cancelOnMovement: 1000,
  onFinish: () => {
    console.log('finished');
    setDefaultSubs();
  }
});
```

This works pretty well, but the only case where I'm having issues is if the long press moves around it successfully triggers, but then releasing the long press does not trigger the "onFinish":

_In the screen recording below, you can see that the "long pressed" message appears and the content changes. But when I release the tap (around the time the scrolling stops) the "finished" message does not appear until I click on the webpage again._

![brave_dS3yfuRz3F](https://github.com/minwork/react/assets/1558019/4cd3c220-1084-4d33-9f6b-b89256de3d1c)

If I keep the tap in the same location without moving it, it works exactly how I would expect.

What setting do I need to change to get this behavior?

**Comment by minwork** [comment]
Hey, when you set `cancelOutsideElement` to `false` cancel event (which triggers `onFinish` callback) will be detected by both element and window listener but it seems that window listener didn't catch that event.

There may a be a few reasons for that:
1. Google chrome mobile device emulator tends to call multiple different events to emulate touch which may trigger strange behaviour. Have you tried if that flow is working when using mouse only or on actual mobile device?
2. Is there any element that could suppress bubbling up `mouseup`, `touchend` or `pointerup` events to `window`?
3. Try out `use-long-press@3.2.1-alpha.1` where I've changed how events are recognised as valid, to check if that affects this situation in any way.

**Reply by derekantrican** [reply]
Thanks for the reply.

1. I have tested this on a mobile device - same results
2. I don't think so - I'm putting this on one of the "most root" elements (I think it is a `<div>` under `<body>`). In my testing I'm also trying to avoid tapping on other parts of the app that would trigger an event (eg a hyperlink or `onClick` event)
3. I'll try that out when I get a chance

**Comment by derekantrican** [comment]
Closing as I've decided this approach isn't going to work for me (even if I get the long press to work the way I want)
